### DISCUSSION TITLE
GSC (Google Search Console) [start] [ERROR] Search Console API Error for domain.com (30days) :  403 Bad Request

### DISCUSSION THREAD

**Discussion by yuvrajsagoo** [towfiqi/serpbear#91] [Q&A]
Hey there Towfiqi, 

Is there any update on this : [start] [ERROR] Search Console API Error for domain.com (30days) :  403 Bad Request.

been trying hard but it isnt working.

**Comment by tristanbh** [comment]
Just to chime in here, I've been running into the same issue. Both through an instance hosted on PikaPods and on my own server. I've verified my GSC API is running, added the user to our GSC account, followed all the steps as outlined in the docs, and still get 403s.

**Reply by yuvrajsagoo** [reply]
> Just to chime in here, I've been running into the same issue. Both through an instance hosted on PikaPods and on my own server. I've verified my GSC API is running, added the user to our GSC account, followed all the steps as outlined in the docs, and still get 403s.

Exactly. I have been trying since January I think or Feb Early. No luck. only 403 Error.

**Reply by gboxxx** [reply]
Same issue here self-hosted on docker.

**Reply by sergiorodenas** [reply]
Same here

**Comment by towfiqi** [comment]
As you know, 403 means permission error. Which means the API key you inserted does not have the permission to make the calls for that specific domain. Its working fine on my installation. 

**Ensure that you have completed the Required Steps:**
1. You have inserted the Service key Email address as a user in in your Google Search Console domain property.
2. You have inserted the `private_key` and the `client_email` values in your env.local file. Or if you are using docker, then follow [this step](https://docs.docker.com/engine/reference/commandline/run/#env) . But beware that since the `private_key`  contains new line, the key might be incompletely stored as described [here](https://github.com/moby/moby/issues/12997).

**Comment by sergiorodenas** [comment]
Got it working. You need to give access to the service account to a "domain property" @ Google Search Console.

**Comment by towfiqi** [comment]
@sergiorodenas Great! Can you please check the documentation and see if I missed describing a step? https://docs.serpbear.com/miscellaneous/integrate-google-search-console

**Comment by everythingdigital** [comment]
are there a way to get URL prefix property not Domain property datas?

**Reply by towfiqi** [reply]
That's not currently possible.

**Comment by towfiqi** [comment]
This issue was caused by SerpBear's inability to add the URL properties of the Google Search Console.  It only supported domain property. The ability to add URL property has been added in v1.0.0 so this issue should not occur anymore. If anyone is having this issue, try updating your SerpBear instance to the latest version.
